src/linear_algebra/prod.lean

```diff
@@ -800,3 +800,29 @@ end
 end graph
 
 end linear_map
+
+/-! ### Projections of submodules -/
+section projection
+
+namespace submodule
+
+variables [semiring R] [add_comm_monoid M] [add_comm_monoid M₂]
+  [module R M] [module R M₂]
+
+/-- The projection of a submodule onto the first component. -/
+def proj_fst (S : submodule R (M × M₂)) : submodule R M := S.map (linear_map.fst R M M₂)
```

Comment:
There are a few lemmas above formulated in terms of `map (fst _ _ _) _` and `map (snd _ _ _) _`. If you add new definitions, then you should reformulate them in terms of `proj_fst` and `proj_snd`.